astro/cli/astro/cmd/cmd.go

```diff
@@ -0,0 +1,291 @@
+/*
+ *  Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package cmd contains the source for the `astro` command line tool
+// that operators use to interact with the project.
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/uber/astro/astro"
+	"github.com/uber/astro/astro/conf"
+	"github.com/uber/astro/astro/logger"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	// silence trace info from terraform/dag by default
+	log.SetOutput(ioutil.Discard)
+}
+
+// AstroCLI is the main CLI program, where flags and state are stored for the
+// running program.
+type AstroCLI struct {
+	stdin  io.Reader
+	stdout io.Writer
+	stderr io.Writer
+
+	project *astro.Project
```

Comment:
`astro.Project` is in a separate package. Once you pass in the config to an astro project, you have to keep your own reference. That's why we have it here.

The only reason we're keeping it here is due to the complicated lifecycle interaction caused by friction with Cobra, which is opinionated about how you structure your CLI project. We're fighting against Cobra because we're trying to dynamically set flags based on config (which could be loaded due to a flag!) so long term we should probably drop Cobra in favour of our own parsing (or more dynamic parsing library).